### PR TITLE
Advanced/Dive into AVM

### PR DESCRIPTION
**Pull request by ramidecodes**
Add content for:
* AVM Overview
* Account keys generation
* R1CS Circuits
----

Other considerations:
* Diagram images are not final (currently just for reference imported from other sources) until we decide on a concrete style to generate all the diagrams for the docs
* Using conventional LaTeX notation for equations, but we may need to update those depending on what we decided to use for all the math notation across the docs.
* We probably need to revisit all the content once the main body of knowledge is implemented in order to do proper back linking between all the different sections and topics covered in the docs. 
* I couldn't find any academic reference to Varuna (modified Marlin)

----
Linked to this Asana task: https://app.asana.com/0/1206873438735612/1207043693511590/f

### CODE REVIEW

**Comment by Zack-Xb** [comment]
I have no further comments apart from what @ghostant-1017 added

File: src/advanced/dive-into-avm/aleo-account-keys.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Aleo Account Key Generation
+
+An Aleo account is similar to accounts on other blockchain platforms but designed with a focus on privacy, leveraging cryptographic techniques unique to the platform. Here's a detailed breakdown of the components of an Aleo account and the processes involved in generating addresses, public keys, and private keys:
+
+## **Account Structure**
+
+- **Private Key**: This is the secret key that an account holder must protect. It is used to sign transactions and prove ownership of the account. In Aleo, the private key is typically a large random number.
```
[the remark this answers is not in the document]
Comment (ramidecodes) [reply src/advanced/dive-into-avm/aleo-account-keys.md:7]: Mmm... I see a compute key in the VM code:
Screenshot 2024-05-13 at 09 22 21

**Comment by ramidecodes** [comment]
I don't know the author of that article. Just added the source bellow the image. But I don't think we'll be using this image in the final version. This is just a place holder, until we define a unified style for all the images and diagrams in the documentation.

**Review by laishawadhwa — COMMENTED** [review]
LGTM

**Pull request merged**